### PR TITLE
Fix BentoML deployer

### PR DESCRIPTION
**Pull request by safoinme**
## Describe changes
I implemented/fixed _ to achieve _.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: examples/bentoml_deployment/steps/predictor.py [hunk 1]
```diff
@@ -33,7 +33,11 @@ def predictor(
     """
     service.start(timeout=10)  # should be a NOP if already started
     for img, data in inference_data.items():
-        prediction = service.predict("predict_ndarray", np.array(data))
+        grayscale_image = np.dot(
+            np.array(data)[..., :3], [0.2989, 0.5870, 0.1140]
+        )
+        grayscale_image = grayscale_image.astype(np.uint8)
+        prediction = service.predict("predict_ndarray", grayscale_image)
```
Comment (fa9r) [inline examples/bentoml_deployment/steps/predictor.py:40]: why can't it predict on the original data directly anymore?

Comment (safoinme) [reply examples/bentoml_deployment/steps/predictor.py:40]: Actually, I was surprised by how it was working from the start! the image we were using for inference where (28, 28, 3), and in the bentoml service validation we were asserting that it was greyscale (28, 28) and it was working somehow

**Comment by safoinme** [comment]
@fa9r  I'm not so sure what exactly was blocking this, but I just went through Bentoml code and update the serve function based on what they have now (deprecating the development function and setting all these variables) and it worked after that!

File: examples/bentoml_deployment/steps/predictor.py (hunk 1 above)
Comment (fa9r) [reply examples/bentoml_deployment/steps/predictor.py:40]: Ah so this is turning it to greyscale? In that case, just add a docstring please so we will remember this in the future 😄 

```
       grayscale_image = np.dot(
            np.array(data)[..., :3], [0.2989, 0.5870, 0.1140]
        )
```

**Review by fa9r — APPROVED** [review]
Ok cool, if it works it works, let's go to main! 😄 🚀

File: examples/bentoml_deployment/steps/predictor.py (hunk 1 above)
Comment (strickvl) [reply examples/bentoml_deployment/steps/predictor.py:40]: @safoinme will you add this?

**Review by strickvl — APPROVED** [review]
LGTM! We should reach out to the people who were having problems with this in Slack once it's merged / released.

**Pull request merged**